### PR TITLE
core: [NFC] Split irdl.py

### PR DESCRIPTION
**Pull request by PapyChacal**

### CODE REVIEW

**Review by AntonLydike — APPROVED** [review]

**Review by georgebisbas — APPROVED** [review]

**Review by superlopuh — APPROVED** [review]
Finally :)

File: xdsl/irdl/common.py [hunk 1]
```diff
@@ -0,0 +1,18 @@
+from enum import Enum
```
Comment (math-fehr) [inline xdsl/irdl/common.py:1]: Should we rename this to `error.py` for now?

**Review by math-fehr — APPROVED** [review]
Doing the lord's work, thanks a lot 🙏

File: xdsl/irdl/common.py (hunk 1 above)
Comment (PapyChacal) [reply xdsl/irdl/common.py:1]: Why not!

**Comment by superlopuh** [comment]
I moved around the branches to merge the split before the RangeConstraint PR, to unblock other changes that would have touched irdl.py

**Pull request merged**